src/infrastructure/storage/tools.ts

```diff
@@ -0,0 +1,31 @@
+import type { EditorTool } from '@/domain/entities/EditorTool';
+import { SubscribableStore } from './abstract/subscribable';
+
+/**
+ * Data stored in the user store
+ */
+export type EditorToolStoreData = {
+  /**
+   * User data
+   */
+  editorTool: EditorTool | null;
+};
+
+/**
+ * Store for the editor tools
+ */
+export class EditorToolStore extends SubscribableStore<EditorToolStoreData> {
```

Comment:
I think, you can get rid of the store for this task. Just load tools from transport every time.

Store is used when we need to keep some data in memory, but there is no reason for that at the moment. We can add it later if needed.

Subscribable Store — is a kind of store that empowers reactivity to the stored data.